rust/src/utils.rs

```diff
@@ -197,22 +197,22 @@ mod tests {
 
 mod ffi {
     use super::*;
+    use crate::ffiutil::*;
     use glib;
+    use glib::translate::*;
     use glib_sys;
     use libc;
     use std::ffi::CString;
     use std::os::unix::io::IntoRawFd;
     use std::ptr;
 
-    use crate::ffiutil::*;
-
     #[no_mangle]
     pub extern "C" fn ror_download_to_fd(
         url: *const libc::c_char,
         gerror: *mut *mut glib_sys::GError,
     ) -> libc::c_int {
-        let url = ffi_view_nullable_str(url).unwrap();
-        match download_url_to_tmpfile(url) {
+        let url: String = unsafe { from_glib_none(url) };
+        match download_url_to_tmpfile(url.as_str()) {
```

Comment:
This and below could just be

```suggestion
        match download_url_to_tmpfile(&url) {
```

no? Obviously not a big deal, just curious if that was on purpose.

Reply:
I have sort of trained myself to use `.as_str()` because in a lot of cases it is just less typing to have a function take `foo: &str` as a parameter rather than go generic with `AsRef<str>`.  (And `&String` and `&str` are not the same thing)

Reply:
Hmm, but `&String` automatically gets coerced to `&str`, right? So you could leave the function parameter as `&str` and `&url` should just work.

Reply:
Hmmm yes it does seem so in a quick test but I'd swear I hit problems with this elsewhere.